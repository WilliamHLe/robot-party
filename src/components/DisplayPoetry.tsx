import React from "react";
import './DisplayPoetry.css';



type DState = {
    title: string
    author: string
    lines: object
}
type DProps = {
    url: string
}

// Henter dikt fra PoetryDB
// Bruker en klasse-komponent
class DisplayPoetry extends React.Component<DProps, DState> {
    state = {title: "", author: "", lines: [""]}

    componentDidMount() {
        fetch(this.props.url)
            .then(response => response.json())
            .then((data) => {
                this.setState({title: data[0].title, author: data[0].author, lines: data[0].lines })
                console.log(data)

            })
        console.log(this.state.title)
    }

    renderContent() {
        if (this.state.title !== "") {
            return (
                <div className="poetry-container">
                    <h3>{this.state.title} av {this.state.author}</h3>
                    <div>{this.state.lines.map((line) => <p>{line}</p>)}</div>
                </div>
            )
        }

        return <h1>Vennligst vent mens diktet hentes...</h1>;
    }

    render() {
        return (
            this.renderContent()
        )
    }
}

export default DisplayPoetry