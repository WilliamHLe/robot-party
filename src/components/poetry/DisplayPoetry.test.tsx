import React from 'react';
import DisplayPoetry from "./DisplayPoetry";
const TestRenderer = require('react-test-renderer');


it('renders correctly when poem is fetched', () => {
    const tree = TestRenderer.create(<DisplayPoetry  url={"https://poetrydb.org/author,title/Emily%20Dickinson;We%20should%20not%20mind%20so%20small%20a%20flower"} />).toJSON();
    expect(tree).toMatchSnapshot();
});