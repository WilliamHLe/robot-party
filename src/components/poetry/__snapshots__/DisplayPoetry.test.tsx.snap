// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly when poem is fetched 1`] = `
<h1>
  Vennligst vent mens diktet hentes...
</h1>
`;
